### ISSUE TITLE
ArrayCollectionHelper - problem with sorting

### ISSUE THREAD

**Issue by f3l1x** [nextras/orm#211]
Hi. 

I've just discovered strange behaviour and I do not know if it's feature or bug.

We have entity `Order` like this:

```
 * @property OneHasMany|OrderItem[] $items   {1:m OrderItem::$order, orderBy=[this->product->manufacturer->name=ASC, name=ASC]}
```

But `OrderItem` has nullable relation to product, it might not exist (some triggers in DB fill order item product).

Well and ArrayCollectionHelper throws exception `Call to a member function hasValue() on null`.

![selection_002](https://cloud.githubusercontent.com/assets/538058/23901737/d4ebb2aa-08be-11e7-8f3d-481027fd67ff.png)

Ofcource it's NULL, relation to product is NULL. What I think is, that this entities should be skipped or second rule should be matched, what do you think? Or maybe some exception with NULL warning? 

@hrach?

**Comment by hrach** [comment]
It's a bug.

**Comment by hrach** [comment]
0 should be returned, therefore the later rules may apply.

**Comment by f3l1x** [comment]
Thanks, I was hoping that it's a bug. :+1:

**Comment by hrach** [comment]
fixed also in branch v2.2. will release stable tag asap, I want take a look on your second bug.
